src/components/molecules/TableGrid/TableGrid.tsx

```diff
@@ -38,21 +39,62 @@ export interface TablesUserListProps {
   leaveText?: string;
   venue: WithId<AnyVenue>;
   venueId: string;
-  template: VenueTemplate;
+  userId: string;
 }
 
-export const TablesUserList: React.FC<TablesUserListProps> = ({
+export const TableGrid: React.FC<TableGridProps> = ({
   venueId,
-  setSeatedAtTable,
-  seatedAtTable,
   customTables,
   defaultTables,
   showOnlyAvailableTables = false,
   TableComponent,
   joinMessage,
   venue,
-  template,
+  userId,
 }) => {
+  const analytics = useAnalytics({ venue });
+  const [seatedAtTable, setSeatedAtTable] = useState<string>();
+
+  useUpdateTableRecentSeatedUsers(
+    VenueTemplate.jazzbar,
+    seatedAtTable && venue?.id
+  );
+
+  useEffect(() => {
+    seatedAtTable && analytics.trackSelectTableEvent();
+  }, [analytics, seatedAtTable]);
+
+  const { joinHuddle, leaveHuddle, inHuddle } = useVideoHuddle();
+
+  const wasPrevAtTable = !!seatedAtTable;
+  const joinTable = useCallback(
+    (table) => {
+      // If the user is already in a huddle and was previously at another
+      // table then make sure to leave the huddle first. This covers the
+      // scenario where a user clicks "join" when already at a different
+      // table
+      if (wasPrevAtTable && table) {
+        leaveHuddle();
+      }
+      joinHuddle(userId, `${venue.id}-${table}`);
+      setSeatedAtTable(table);
+    },
+    [wasPrevAtTable, joinHuddle, userId, venue.id, leaveHuddle]
+  );
+
+  const leaveTable = useCallback(async () => {
+    if (!userId) return;
+
+    await unsetTableSeat(userId, { venueId: venue.id });
```

Comment:
[OOS] Should probably refactor those functions to be with more uniformed interface for DX, like

```suggestion
    await unsetTableSeat({ userId, spaceId });
```